### DISCUSSION TITLE
User Metrics

### DISCUSSION THREAD

**Discussion by vogtjob** [NUWCDIVNPT/stig-manager#1233] [Q&A]
Hi, I was asked if there was a way to collect user metrics.  For example, Is it possible to collect metrics on the number of assets uploaded by an individual or the amount of rules submitted by a user?

**Comment by cd-rite** [comment]
Hi @vogtjob We do not collect/calculate high-level metrics by user in the same way we do for Asset/STIG Reviews.  
However, the `/collections/{collectionId}/reviews` endpoint does support a `userId` query parameter that could be used to fetch all the Reviews that were evaluated by a particular user in that collection.  

We don't have plans to offer a view of this in the UI on our immediate roadmap, but it would certainly be possible to write your own client that made use of this endpoint to build that kind of report. 

We do not track who created Assets or STIG assignments internal to the application. However, we do create a log entry with the username for every request to the API, and those requests that create/assign assets would be fairly distinct (either a POST or PUT to the `/assets` endpoint).  If you are using a log aggregator like Splunk to capture those logs, you could probably build a report that could give you the metrics you are looking for. 

This isn't to say we will *never* capture those kinds of stats, but it's not our current priority.

**Reply by cd-rite** [reply]
Hi @vogtjob Our latest release includes a feature you asked about here!

**(API) feat: Added resultEngine, user, statusUser columns to Detail metrics exports.**

With those new columns in the export, you can get metrics about user contributions to the current state of your Collection, aggregated using the same criteria as the other metrics.  They are split up by statusUser (User that last set the status of a review) and "user" (User who last set the "evaluation" portion of a review). If you are using automated tools, there is also a ResultEngine column aggregating contributions from those different tools/versions.
